File: src/main/java/br/ufba/jnose/entities/Projeto.java
package br.ufba.jnose.entities;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import java.io.Serializable;
import java.util.List;
import java.util.Objects;

@Entity
public class Projeto implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue
    private Long id;

    private String name;

    private String path;

    private String url;

    private Integer stars;

    private String junitVersion;

    @OneToMany
    private List<Projeto> projetos;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public Integer getStars() {
        return stars;
    }

    public void setStars(Integer stars) {
        this.stars = stars;
    }

    public String getJunitVersion() {
        return junitVersion;
    }

    public void setJunitVersion(String junitVersion) {
        this.junitVersion = junitVersion;
    }

    public List<Projeto> getProjetos() {
        return projetos;
    }

    public void setProjetos(List<Projeto> projetos) {
        this.projetos = projetos;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Projeto projeto = (Projeto) o;
        return id.equals(projeto.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

    @Override
    public String toString() {
        return "Projeto{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", path='" + path + '\'' +
                ", url='" + url + '\'' +
                ", stars=" + stars +
                ", junitVersion='" + junitVersion + '\'' +
                '}';
    }
}

File: src/main/java/io/github/arieslab/entities/Projeto.kt
package io.github.arieslab.entities

import jakarta.persistence.*
import java.io.Serializable
import java.util.*

@Entity
class Projeto : Serializable {

    @Id
    @GeneratedValue
    var id: Long? = null

    var name: String = ""

    var path: String = ""

    var url: String = ""

    var stars: Int = 0

    var junitVersion: String = ""

    var dateUpdate: Date? = null

    @OneToMany
    var testSmell: List<TestSmell>? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Projeto) return false
        return id == other.id
    }

    override fun hashCode() = Objects.hash(id)

    override fun toString() = "Projeto{id=$id, name='$name', path='$path', url='$url', stars=$stars, junitVersion='$junitVersion'}"
}
